routes/users_routes.js

```diff
@@ -409,116 +475,100 @@ router.get('/logout', function(req, res) {
   res.redirect('/');
 });
 
-router.get('/view/:id', function(req, res) {
-  User.findById(req.params.id, function(err, user) {
+router.get('/view/:id', async function(req, res) {
+  var user;
+  try {
+    user = await User.findById(req.params.id);
```

Comment:
I think this should be one query; there are a lot of other reasons queries could fail. The second query right below this isn't wrapped in a try-catch.